spyder_notebook/notebookplugin.py

```diff
@@ -387,6 +387,11 @@ def close_client(self, index=None, client=None, save=False):
             client.shutdown_kernel()
         client.close()
 
+        # Delete notebook file if it is in temporary directory
+        filename = client.get_filename()
+        if filename.startswith(get_temp_dir()):
+            os.remove(filename)

```

Comment:
Let's enclose this in a `try/except Exception` because there's no need to report an error in case this fails.